jmh-benchmarks/src/main/java/org/apache/kafka/jmh/acl/AclAuthorizerBenchmark.java

```diff
@@ -0,0 +1,137 @@
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+import scala.collection.JavaConverters;
+import scala.collection.immutable.TreeMap;
+import scala.math.Ordering;
+
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+@State(Scope.Benchmark)
+@Fork(value = 1)
+@Warmup(iterations = 5)
+@Measurement(iterations = 15)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+
+public class AclAuthorizerBenchmark {
+    @Param({"5000", "10000", "50000"})
+    private int resourceCount;
+    //no. of. rules per resource
+    @Param({"5", "10", "15"})
+    private int aclCount;
+
+    private AclAuthorizer aclAuthorizer = new AclAuthorizer();
+    private KafkaPrincipal principal = new KafkaPrincipal(KafkaPrincipal.USER_TYPE, "test-user");
+
+    @Setup(Level.Trial)
+    public void setup() throws Exception {
+        setFieldValue(aclAuthorizer, AclAuthorizer.class.getDeclaredField("aclCache").getName(),
+            prepareAclCache());
+    }
+
+    private void setFieldValue(Object obj, String fieldName, Object value) throws Exception {
+        Field field = obj.getClass().getDeclaredField(fieldName);
+        field.setAccessible(true);
+        field.set(obj, value);
+    }
+
+    private TreeMap<ResourcePattern, VersionedAcls> prepareAclCache() {
+        Map<ResourcePattern, Set<AclEntry>> aclEntries = new HashMap<>();
+        for (int resourceId = 0; resourceId < resourceCount; resourceId++) {
+
+            ResourcePattern resource = new ResourcePattern(
+                (resourceId % 10 == 0) ? ResourceType.GROUP : ResourceType.TOPIC,
+                "resource-" + resourceId,
+                (resourceId % 5 == 0) ? PatternType.PREFIXED : PatternType.LITERAL);
+
+            Set<AclEntry> entries = aclEntries.computeIfAbsent(resource, k -> new HashSet<>());
+
+            for (int aclId = 0; aclId < aclCount; aclId++) {
+                AccessControlEntry ace = new AccessControlEntry(principal.toString() + aclId,
+                    "*", AclOperation.READ, AclPermissionType.ALLOW);
+                entries.add(new AclEntry(ace));
+            }
+        }
+
+        TreeMap<ResourcePattern, VersionedAcls> aclCache = new TreeMap<>(new ResourceOrdering());
+        for (Map.Entry<ResourcePattern, Set<AclEntry>> entry : aclEntries.entrySet()) {
+            aclCache = aclCache.updated(entry.getKey(),
+                new VersionedAcls(JavaConverters.asScalaSetConverter(entry.getValue()).asScala().toSet(), 1));
+        }
+
+        return aclCache;
+    }
+
+    @TearDown(Level.Trial)
+    public void tearDown() {
+        aclAuthorizer.close();
+    }
+
+    @Benchmark
+    public void testAclsIterator() {
+        aclAuthorizer.acls(AclBindingFilter.ANY);
+    }
+
+    private static class ResourceOrdering implements Ordering<ResourcePattern> {
```

Comment:
AclAuthorizer is not a public class, so it may be ok to make this class public in AclAuthorizer instead of duplicating it here.